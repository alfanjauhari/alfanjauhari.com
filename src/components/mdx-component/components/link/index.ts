export * from './link';
export * from './link.styles';
