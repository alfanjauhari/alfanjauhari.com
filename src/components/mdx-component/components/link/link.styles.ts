import { styled } from '@/theme';
import Link from 'next/link';

export const StyledInternalLink = styled(Link, {
  color: '$blue6',
  '&:hover': {
    color: '$blue8',
    transitionDuration: '300ms',
  },
});

export const StyledExternalLink = styled('a', StyledInternalLink);
