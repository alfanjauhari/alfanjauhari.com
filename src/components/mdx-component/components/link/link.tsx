import { AnchorHTMLAttributes } from 'react';
import { StyledExternalLink, StyledInternalLink } from './link.styles';

export type LinkProps = AnchorHTMLAttributes<HTMLAnchorElement>;

export function Link({ href, children, ...props }: LinkProps) {
  const isInternalLink = href?.startsWith('/') || href?.startsWith('#');

  if (isInternalLink && href) {
    return (
      <StyledInternalLink href={href} {...props}>
        {children}
      </StyledInternalLink>
    );
  }

  return (
    <StyledExternalLink target="_blank" href={href} rel="noreferrer" {...props}>
      {children}
    </StyledExternalLink>
  );
}
