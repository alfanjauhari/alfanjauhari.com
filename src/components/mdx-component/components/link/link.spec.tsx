import { render, screen } from '@testing-library/react';
import { Link } from './link';

describe('MDXComponent/Link', () => {
  it('should render successfully', () => {
    render(<Link href="/">Home</Link>);

    expect(
      screen.getByRole('link', {
        name: 'Home',
      }),
    ).toBeVisible();
  });

  it('should render external link component if href is an external url', () => {
    render(<Link href="https://google.com">Google</Link>);

    expect(
      screen.getByRole('link', {
        name: 'Google',
      }),
    ).toHaveAttribute('target', '_blank');
  });

  it('should match snapshot', () => {
    render(<Link href="/">Home</Link>);

    expect(
      screen.getByRole('link', {
        name: 'Home',
      }),
    ).toMatchSnapshot();
  });
});
