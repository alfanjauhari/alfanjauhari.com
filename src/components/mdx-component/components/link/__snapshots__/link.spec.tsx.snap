// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXComponent/Link should match snapshot 1`] = `
<a
  class="c-ktVJdH"
  href="/"
>
  Home
</a>
`;
